dbeam-core/src/test/java/com/spotify/dbeam/options/InputAvroSchemaTest.java

```diff
@@ -0,0 +1,190 @@
+import org.junit.Test;
+
+public class InputAvroSchemaTest {
+
+  private static File avroSchemaFile;
+  private static Path avroSchemaFilePath;
+  private static String avroSchemaFilePathStr;
+
+  @BeforeClass
+  public static void beforeAll() throws IOException {
+    String jsonSchema =
+        "{\n"
+        + "  \"name\": \"Show\",\n"
+        + "  \"doc\": \"Record description\",\n"
+        + "  \"namespace\": \"v2\",\n"
+        + "  \"type\": \"record\",\n"
+        + "  \"fields\": [\n"
+        + "    {\n"
+        + "      \"name\": \"field1\",\n"
+        + "      \"type\": \"string\",\n"
+        + "      \"doc\": \"Field1 description\"\n"
+        + "    },\n"
+        + "    {\n"
+        + "      \"name\": \"field2\",\n"
+        + "      \"type\": \"string\",\n"
+        + "      \"doc\": \"Field2 description\"\n"
+        + "    }\n"
+        + "  ]\n"
+        + "}\n";
+
+    avroSchemaFile = createTestAvroSchemaFile(jsonSchema);
+    avroSchemaFilePath = avroSchemaFile.toPath();
+    avroSchemaFilePathStr = avroSchemaFilePath.toString();
+  }
+
+  private static File createTestAvroSchemaFile(final String jsonSchema) throws IOException {
+    File newTempFile = File.createTempFile("dataType1", ".avsc");
+    newTempFile.deleteOnExit();
+    Files.write(
+        newTempFile.toPath(),
+        jsonSchema.getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.CREATE);
+    return newTempFile;
+  }
+
+  @AfterClass
+  public static void afterAll() throws IOException {
+    Files.delete(avroSchemaFile.toPath());
+  }
+
+  @Test
+  public void checkReadAvroSchema() throws IOException {
+
+    JdbcExportPipelineOptions options =
+        PipelineOptionsFactory.create().as(JdbcExportPipelineOptions.class);
+    options.setAvroSchemaFilePath(avroSchemaFilePathStr);
+
+    Assert.assertEquals(avroSchemaFilePathStr, options.getAvroSchemaFilePath());
+
+    final Schema inputSchema = BeamJdbcAvroSchema.parseInputAvroSchemaFile(avroSchemaFilePathStr);
+
+    Assert.assertEquals("Record description", inputSchema.getDoc());
+    Assert.assertEquals("Field1 description", inputSchema.getField("field1").doc());
+    Assert.assertEquals("Field2 description", inputSchema.getField("field2").doc());
+  }
+
+  @Test
+  public void checkReadAvroSchemaWithEmptyParameter() throws IOException {
+
+    String path = "";
+    JdbcExportPipelineOptions options =
+        PipelineOptionsFactory.create().as(JdbcExportPipelineOptions.class);
+    options.setAvroSchemaFilePath(path);
+
+    Assert.assertEquals(path, options.getAvroSchemaFilePath());
+    Assert.assertEquals(
+        Optional.empty(), BeamJdbcAvroSchema.parseOptionalInputAvroSchemaFile(path));
+  }
+
+  @Test
+  public void checkReadAvroSchemaWithNullParameter() throws IOException {
+
+    String path = null;
+    JdbcExportPipelineOptions options =
+        PipelineOptionsFactory.create().as(JdbcExportPipelineOptions.class);
+    options.setAvroSchemaFilePath(path);
```

Comment:
Another way to test this is to use `fromArgs`.

i.e.

```
PipelineOptionsFactory.fromArgs().create(); // e.g. no args
PipelineOptionsFactory.fromArgs("--avroSchemaFilePath=").create(); // empty avroSchemaFilePath
```

I believe this is better because using command line arguments is what is in the DBeam documentation (README). So we can use these tests as examples and validation of the offered functionality.

Reply:
Ah, I now saw below a few tests using `QueryBuilderArgsTest.commandLineToOptions`. If we agree on the above comment, we can remove the tests that use `setAvroSchemaFilePath()` and only leave the ones using `commandLineToOptions()`